test/test_roots.cpp

```diff
@@ -649,11 +651,121 @@ void test_failures()
+         fn_push_back_x_scaled(-x);
+      };
+
+      fn_push_back_pm(v, T(0.0));
+      fn_push_back_pm(v, std::numeric_limits<T>::denorm_min());
+      fn_push_back_pm(v, (std::numeric_limits<T>::min)());
+      fn_push_back_pm(v, std::numeric_limits<T>::epsilon());
+      const int test_exp_range = 5;
+      for (int i = -test_exp_range; i < (test_exp_range + 1); ++i) {
+         const int exponent = i * 3;
+         const T x = std::ldexp(1.0, exponent);
+         fn_push_back_pm(v, x);
+      }
+      fn_push_back_pm(v, (std::numeric_limits<T>::max)());
+
+      // Real_concept doesn't have infinity
+      if (std::numeric_limits<T>::has_infinity) {
+         fn_push_back_pm(v, std::numeric_limits<T>::infinity());
+      }
+
+      std::sort(v.begin(), v.end());
+
+      const int num_zeros = std::count(v.begin(), v.end(), T(0.0));
+      BOOST_CHECK_LE(2, num_zeros);  // Positive and negative zero 
+
+      return v;
+   };
+
+   // Test for all combinations from the ladder
+   auto v = fn_create_test_ladder();
+   for (const T& x_i: v) {
+      for (const T& x_j: v) {
+         const T x_mid_ij = boost::math::tools::detail::Bisection::calc_midpoint(x_i, x_j);
+
+         const T x_lo = (std::min)(x_i, x_j);
+         const T x_hi = (std::max)(x_i, x_j);
+
+         BOOST_CHECK_LE(x_lo, x_mid_ij);
+         BOOST_CHECK_LE(x_mid_ij, x_hi);
+      }
+   }
+}
+
+void test_bisect_all_cases() {
+   test_bisect<float>();
+   test_bisect<double>();
+   test_bisect<long double>();
+   test_bisect<boost::math::concepts::real_concept>();
+   test_bisect<boost::multiprecision::cpp_bin_float_50>();
+   test_bisect<boost::multiprecision::cpp_bin_float_100>();
+   test_bisect<boost::multiprecision::cpp_dec_float_50>();
+   test_bisect<boost::multiprecision::cpp_dec_float_100>();
+}
+
+void test_count() {
+   const auto fn_newton = [](std::uintmax_t& i_max) {
+      return boost::math::tools::newton_raphson_iterate(
+         [](double x) { return std::make_pair(x * x - 3, 2 * x); },
+         10.0  /* x_initial */,
+         0.0   /* bound lower*/,
+         100.0 /* bound upper */, 52, i_max);
+   };
+
+   // Find the minimum number of iterations to find the solution
+   std::uintmax_t iter_find_solution = 100;
+   fn_newton(iter_find_solution);
+   BOOST_CHECK_EQUAL(iter_find_solution, std::uintmax_t(8));  // Confirm is 8
+
+   for (std::uintmax_t count = 0; count < (iter_find_solution * 2); count++) {
+      std::uintmax_t iters = count;
+      if (iter_find_solution <= count) {  // Finds correct answer
+         using std::sqrt;
+         BOOST_CHECK_EQUAL(fn_newton(iters), sqrt(3.0));
+         BOOST_CHECK_EQUAL(iters, iter_find_solution);
+      }
+      else {  // Throws error for running out of iterations
+         BOOST_CHECK_THROW(fn_newton(iters), boost::math::evaluation_error);
+      }
+   }
+}
+
 BOOST_AUTO_TEST_CASE( test_main )
 {
 
    test_beta(0.1, "double");
 
+   test_count();
+   test_bisect_all_cases();
+#if 0
```

Comment:
Do the following tests fail with this new method?

Reply:
I accidentally left this in. Good catch.